lib/action_subscriber/rabbit_connection.rb

```diff
@@ -3,57 +3,40 @@
 module ActionSubscriber
   module RabbitConnection
     SUBSCRIBER_CONNECTION_MUTEX = ::Mutex.new
-    PUBLISHER_CONNECTION_MUTEX = ::Mutex.new
     NETWORK_RECOVERY_INTERVAL = 1.freeze
 
-    def self.publisher_connected?
-      publisher_connection.try(:connected?)
-    end
-
-    def self.publisher_connection
+    def self.setup_connection(name, settings)
       SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        return @publisher_connection if @publisher_connection
-        @publisher_connection = create_connection
-      end
-    end
-
-    def self.publisher_disconnect!
-      SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        if @publisher_connection && @publisher_connection.connected?
-          @publisher_connection.close
-        end
-
-        @publisher_connection = nil
+        fail ArgumentError, "a #{name} connection already exists" if subscriber_connections[name]
+        subscriber_connections[name] = create_connection(settings)
       end
     end
 
     def self.subscriber_connected?
-      subscriber_connection.try(:connected?)
+      subscriber_connections.all?{|_name, connection| connection.connected?}
     end
 
-    def self.subscriber_connection
+    def self.subscriber_disconnect!
       SUBSCRIBER_CONNECTION_MUTEX.synchronize do
-        return @subscriber_connection if @subscriber_connection
-        @subscriber_connection = create_connection
+        subscriber_connections.each{|_name, connection| connection.close}
+        @subscriber_connections = []
```

Comment:
seems odd that we would set this to `{}` on initialization and then `[]1 on disconnect; maybe we should include concurrent-ruby and use Concurrent::Map in both cases?